### PR TITLE
Implemented @HeaderMap

### PR DESCRIPTION
**Pull request by venilnoronha**
I've created `HeaderMap` which supports parameters of type `Map` ~~and `Multimap`~~. ~~`Multimap` enables one to specify multiple header values for the same name.~~ See #1736 for more info.

I've signed the ICLA. Please review and merge.

Thanks,
Venil

### CODE REVIEW

File: retrofit/pom.xml [hunk 1]
```diff
@@ -32,6 +32,11 @@
     </dependency>
 
     <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+    </dependency>
```
Comment (JakeWharton) [inline retrofit/pom.xml]: We can not depend on Guava

Comment (venilnoronha) [inline retrofit/pom.xml]: Oh okay. Should we have our own `MultiMap`? Or should we support `Map<String, Collection<?>>` instead?

Comment (JakeWharton) [inline retrofit/pom.xml]: Just ignore that case for now, we're going to look into it as part of #626.

Comment (venilnoronha) [inline retrofit/pom.xml]: Sure, thanks!

**Comment by venilnoronha** [comment]
Fixed checkstyle issues, squashed.

**Comment by JakeWharton** [comment]
Looks great at a quick glance! I'll do a full review later.

**Comment by venilnoronha** [comment]
Sure, thank you!

File: retrofit/src/main/java/retrofit2/http/HeaderMap.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.util.Map;
+
+/**
+ * Adds headers specified in the {@link Map}.
+ * <p>
+ * Values are converted to strings using {@link String#valueOf(Object)}.
```
Comment (JakeWharton) [inline retrofit/src/main/java/retrofit2/http/HeaderMap.java]: This is false, but I suspect this is copy/paste from other map annotations so I'll fix them all once this is merged.

**Comment by JakeWharton** [comment]
Merged as adfe99884bfc01dd3a1164a768fef96d6550208b. Thanks again!

**Comment by venilnoronha** [comment]
Thanks a lot @JakeWharton!

**Pull request closed without merging**